programs/bpf_loader/src/syscalls.rs

```diff
@@ -2976,12 +2982,26 @@ where
                 invoke_context,
             )?;
             {
-                let mut account = account.borrow_mut();
-                account.copy_into_owner_from_slice(caller_account.owner.as_ref());
-                account.set_data_from_slice(caller_account.data);
-                account.set_lamports(*caller_account.lamports);
-                account.set_executable(caller_account.executable);
-                account.set_rent_epoch(caller_account.rent_epoch);
+                borrowed_account
+                    .set_lamports(*caller_account.lamports)
+                    .map_err(SyscallError::InstructionError)?;
+                borrowed_account
+                    .set_data(caller_account.data)
+                    .map_err(SyscallError::InstructionError)?;
+                borrowed_account
+                    .set_executable(caller_account.executable)
+                    .map_err(SyscallError::InstructionError)?;
+                borrowed_account
+                    .set_owner(caller_account.owner.as_ref())
+                    .map_err(SyscallError::InstructionError)?;
+                drop(borrowed_account);
+                let account = invoke_context
```

Comment:
why are we setting rent_epoch differently here then the rest?

Reply:
`BorrowedAccount` is the ABIv2 interface, and that does not have rent_epoch anymore.
And yes, what do do with the `rent_epoch` field is an open question in #25899 as well.